### PR TITLE
Posts pagination

### PR DESCRIPTION
**Pull request by JiejayLan**
### Issues
#42 

### UI Design
- `loader.gif` added to create loading screen

image

### Overview of PR
- [x] create a pagination component under home page.
- [x] create a dropdown component for change pageSize under home page
  -  pageSize stands for how posts to display on the home page.
- [x] Improve some coding logic in allPost component
- [x] add stories for both components
- [x] add tests for both components(only write a simple test)

### Technical Changes
- add new library for create a [pagination](https://www.npmjs.com/package/material-ui-flat-pagination) 
- add new library ```MockAdapter``` to mock axios in storybook 

### Architecture Changes
- Replace all es5 function with es6 arrow function

### CODE REVIEW

File: src/components/Pagination/Pagination.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  const handlePageSizeChange = (event) => {
+    const newPageSize = event.target.value;
+    dispatch({
+      type: 'changePage',
+      newPage: { pageSize: newPageSize, currentPage: 1 },
+    });
+  };
+
+  return (
+    <MuiThemeProvider theme={theme}>
```
Comment (nchen000) [inline src/components/Pagination/Pagination.js]: Can you explain why you create another `theme`?

File: src/components/Pagination/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import CssBaseline from '@material-ui/core/CssBaseline';
+import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
+import PaginationComp from 'material-ui-flat-pagination';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import Grid from '@material-ui/core/Grid';
+import useStyles from './style';
+import { useStateValue } from '../StateProvider';
```
Comment (ethen001) [inline src/components/Pagination/Pagination.js]: You should use named imports for material-UI. It serves as refactoring but also as bundle-size optimization. Please read [this](https://material-ui.com/guides/minimizing-bundle-size/).

File: src/components/Pagination/Pagination.js (hunk 1 above)
Comment (JiejayLan) [reply src/components/Pagination/Pagination.js]: I dont think i need to use theme

File: src/components/Pagination/Pagination.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            <InputLabel className={classes.formTitle} id="pageSize-input">Per Page</InputLabel>
+            <Select
+              id="pageSize-dropdown"
+              value={pageSize}
+              onChange={handlePageSizeChange}
+            >
+              <MenuItem value={5}>5</MenuItem>
+              <MenuItem value={10}>10</MenuItem>
+              <MenuItem value={20}>20</MenuItem>
+              <MenuItem value={30}>30</MenuItem>
```
Comment (ethen001) [inline src/components/Pagination/Pagination.js]: You should put this in an array and map the values to the `<MenuItem>`. That way code is DRY and is easier to make changes in the future

**Pull request merged**